app/backend/src/couchers/servicers/donations.py

```diff
@@ -0,0 +1,167 @@
+import json
+import logging
+
+import grpc
+import stripe
+
+from couchers import errors, urls
+from couchers.config import config
+from couchers.db import session_scope
+from couchers.models import Invoice, OneTimeDonation, RecurringDonation, User
+from couchers.sql import couchers_select as select
+from couchers.tasks import send_donation_email
+from couchers.utils import now
+from proto import donations_pb2, donations_pb2_grpc, stripe_pb2_grpc
+from proto.google.api import httpbody_pb2
+
+logger = logging.getLogger(__name__)
+
+
+class Donations(donations_pb2_grpc.DonationsServicer):
+    def __init__(self):
+        self.enabled = config["ENABLE_DONATIONS"]
+
+        if self.enabled:
+            stripe.api_key = config["STRIPE_API_KEY"]
+
+    def InitiateDonation(self, request, context):
+        if not self.enabled:
+            context.abort(grpc.StatusCode.UNAVAILABLE, errors.DONATIONS_DISABLED)
+
+        with session_scope() as session:
+            user = session.execute(select(User).where(User.id == context.user_id)).scalar_one()
+
+            if request.amount < 2:
+                # we don't want to waste *all* the donations on processing fees
+                context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.DONATION_TOO_SMALL)
+
+            if not user.stripe_customer_id:
+                # create a new stripe id for this user
+                customer = stripe.Customer.create(
+                    email=user.email,
+                    # metadata allows us to store arbitrary metadata for ourselves
+                    metadata={"user_id": user.id},
+                )
+                user.stripe_customer_id = customer.id
+                # commit since we only ever want one stripe customer id per user, so if the rest of this api call fails, this will still be saved in the db
+                session.commit()
+
+            if request.recurring:
+                item = {
+                    "price": config.get("STRIPE_RECURRING_PRODUCT_ID"),
```

Comment:
Use `config["STRIPE_RECURRING_PRODUCT_ID"]` to raise an error on the backend side instead of passing `null` to the stripe API.

More occurrences of .get() in this merge request could be converted as well.